src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -636,12 +1271,69 @@ void VM_Version::get_processor_features() {
     _features = feature_flags();
+    _glibc_features = glibc_flags();
     // Logical processors are only available on P4s and above,
     // and only if hyperthreading is available.
     _logical_processors_per_package = logical_processor_count();
     _L1_data_cache_line_size = L1_line_size();
   }
 
+  if (ShowCPUFeatures) {
+    tty->print_cr("This machine's CPU features are: -XX:CPUFeatures=0x" UINT64_FORMAT_X ",0x" UINT64_FORMAT_X, _features, _glibc_features);
+  }
+
+  assert(!CPUFeatures == FLAG_IS_DEFAULT(CPUFeatures), "CPUFeatures parsing");
+  uint64_t       features_expected =   CPU_MAX - 1;
+  uint64_t glibc_features_expected = GLIBC_MAX - 1;
+  if (CPUFeatures) {
+    uint64_t GLIBCFeatures_x64;
+    uint64_t   CPUFeatures_x64 = CPUFeatures_parse(CPUFeatures, GLIBCFeatures_x64);
+    uint64_t       features_missing =   CPUFeatures_x64 & ~      _features;
+    uint64_t glibc_features_missing = GLIBCFeatures_x64 & ~_glibc_features;
+    if (features_missing || glibc_features_missing) {
+      char buf[512];
+      int res = jio_snprintf(
+                  buf, sizeof(buf),
+                  "Specified -XX:CPUFeatures=0x" UINT64_FORMAT_X ",0x" UINT64_FORMAT_X
+                  "; this machine's CPU features are 0x" UINT64_FORMAT_X ",0x" UINT64_FORMAT_X
+                  "; missing features of this CPU are 0x" UINT64_FORMAT_X ",0x" UINT64_FORMAT_X " ",
+                  CPUFeatures_x64, GLIBCFeatures_x64,
+                  _features, _glibc_features,
+                  features_missing, glibc_features_missing);
+      assert(res > 0, "not enough temporary space allocated");
+      // insert_features_names() does crash for undefined too high-numbered features.
+      insert_features_names(buf + res , sizeof(buf) - res ,       features_missing & (  CPU_MAX - 1));
+      int res2 = res + strlen(buf + res);
+      insert_features_names(buf + res2, sizeof(buf) - res2, glibc_features_missing & (GLIBC_MAX - 1));
+      if (buf[res]) {
+        assert(buf[res] == ',', "unexpeced VM_Version::insert_features_names separator instead of ','");
+        buf[res] = '=';
+      }
+      vm_exit_during_initialization(buf);
+    }
+          _features =   CPUFeatures_x64;
+    _glibc_features = GLIBCFeatures_x64;
+
+    if (ShowCPUFeatures) {
+      if (ignore_glibc_not_using) {
+	tty->print_cr("CPU features are being kept intact as requested by -XX:CPUFeatures=ignore");
+      } else {
+	tty->print_cr("CPU features being used are: -XX:CPUFeatures=0x" UINT64_FORMAT_X ",0x" UINT64_FORMAT_X, _features, _glibc_features);
+      }
+    }
+  }
+
+#if !INCLUDE_CPU_FEATURE_ACTIVE && !INCLUDE_LD_SO_LIST_DIAGNOSTICS
+  if (!verify_glibc_by_reexec) {
+          features_expected =       _features;
+    glibc_features_expected = _glibc_features;
+  }
+#endif
+  if (!ignore_glibc_not_using) {
+    glibc_not_using(      features_expected & ~      _features,
+		    glibc_features_expected & ~_glibc_features);
+  }
```

Comment:
I'm trying the code, and getting

```
anton@mercury:~/proj/crac$ ./build/linux-x86_64-server-release/images/jdk/bin/java -XX:+ShowCPUFeatures -XX:CPUFeatures=generic -version
This machine's CPU features are: -XX:CPUFeatures=0x61805fdfbff,0x1e6
CPU features being used are: -XX:CPUFeatures=0x200000080d7,0x0
openjdk version "17-internal" 2021-09-14
OpenJDK Runtime Environment (build 17-internal+0-adhoc..crac)
OpenJDK 64-Bit Server VM (build 17-internal+0-adhoc..crac, mixed mode, sharing)
```

I would expect re-exec, but for some reason it was not perfromed?

Reply:
That was a regression/bug, thanks, fixed.